sql/sql_table.cc

```diff
@@ -8473,11 +8473,21 @@ mysql_prepare_alter_table(THD *thd, TABLE *table,
     }
   }
 
-  if (table->versioned() && !(alter_info->flags & ALTER_DROP_SYSTEM_VERSIONING) &&
-      new_create_list.elements == VERSIONING_FIELDS)
+  if (table->versioned())
```

Comment:
I would appreciate a lot if you could try to move this new code into a helper function, the size of `mysql_prepare_alter_table` is already astronomical.

Reply:
Did it with a new version